### DISCUSSION TITLE
how to fix "curly brace require a decimal"?

### DISCUSSION THREAD

**Discussion by SMods11** [rust-lang/regex#886] [Q&A] [answered]
#### What version of regex are you using?
1.6.0

#### Describe the bug at a high level.

my regex `\\"vault\\"\\:[\\"]?(\\{.*?data.*?\\})` compile with nodejs but seem to have a problem with this library.

#### What are the steps to reproduce the behavior?

```
use regex::Regex;

fn main() {
    let regex = Regex::new(r#"\\"vault\\"\\:[\\"]?(\\{.*?data.*?\\})"#).unwrap();
    let string = r#"roller":{"vault":"{\\"data\\":\\"nac2xYOoNouaNXgx11iiwYLW\\",\\"iv\\":\\"0/h9/3zXRzIALeXu6tHyRA==\\",\\"salt\\":\\"R7eS1IV7k6maUUNivx86OHT/hHhUFpXZgW3Ip1vFp+I=\\"}"}"#;

    let result = regex.captures_iter(string);

    for mat in result {
        println!("{:?}", mat);
    }
}
```

#### What is the actual behavior?

```
hread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Syntax(
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
regex parse error:
    \\"vault\\"\\:[\\"]?(\\{.*?data.*?\\})
                            ^
error: repetition quantifier expects a valid decimal
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
)', src\main.rs:4:73
```

#### What is the expected behavior?

`"vault":"{\\"data\\":\\"nac2xYOoNouaNXgx11iiwYLW\\",\\"iv\\":\\"0/h9/3zXRzIALeXu6tHyRA==\\",\\"salt\\":\\"R7eS1IV7k6maUUNivx86OHT/hHhUFpXZgW3Ip1vFp+I=\\"}`

**Comment by SMods11** [comment, marked as answer]
fixed it with :
```let regex = Regex::new("\"vault\":[\"]?(\\{.*?data.*?\\})").unwrap();```

sorry with the issue

**Comment by BurntSushi** [comment]
> compile with nodejs but seem to have a problem with this library

Note that just because a regex is valid for some other regex engine does not mean it's a bug if it doesn't compile in this regex engine. Basically all regex engines have their own flavor despite generally being very similar.

I would also recommend that you use raw strings. For example, they let you write `r"\w"` instead of needing to write `"\\w"`. With normal strings, you have to contend with both Rust's string literal syntax and the regex syntax. But with raw strings you only need to worry about raw strings.

Pretty much all of the examples in the docs use raw strings for this reason. It's even explicitly called out: https://docs.rs/regex/latest/regex/#example-find-a-date
